Others/TopKWords.java

```diff
@@ -0,0 +1,92 @@
+import  java.io.*;
+import  java.util.*;
+
+/* display the most frequent K words in the file and the times it appear 
+    in the file – shown in order (ignore case and periods) */
+
+public   class  TopKWords {
+    public static void main(String[] a) {
+        // you can replace the filePath with yours
+        CountWords cw = new CountWords("/Users/lisanaaa/Desktop/words.txt");
+        Map<String, Integer> dictionary = cw.getDictionary();  // get the words dictionary: {word: frequency}
+
+        // we change the map to list for convenient sort
+        List<Map.Entry<String, Integer>> list = new ArrayList<>(dictionary.entrySet());
+
+        // sort by lambda valueComparator
+        list.sort(Comparator.comparing(
+                m -> m.getValue())
+        );
+
+        Scanner input = new Scanner(System.in);
+        Integer k = new Integer(input.nextLine());
+        while (k > list.size()) {
+            System.out.println("Retype a number, your number is too large");
+            input = new Scanner(System.in);
+            k = new Integer(input.nextLine());
+        }
+        for (int i = 0; i<k; i++) {
+            System.out.println(list.get(list.size()-i-1));
+        }
+    }
+}
+
+class  CountWords {
```

Comment:
Put this class in another file. Or Write it as inline class. Remove the unnecessary spaces between 'class' and 'CountWords'